vms/platformvm/state.go

```diff
@@ -97,23 +101,30 @@ func (vm *VM) enqueueStaker(db database.Database, subnetID ids.ID, stakerTx *Tx)
 	prefixStart := []byte(fmt.Sprintf("%s%s", subnetID, startDBPrefix))
 	prefixStartDB := prefixdb.NewNested(prefixStart, db)
 
-	p := wrappers.Packer{MaxSize: wrappers.LongLen + wrappers.ByteLen + hashing.HashLen}
-	p.PackLong(uint64(staker.StartTime().Unix()))
-	p.PackByte(priority)
-	p.PackFixedBytes(stakerID[:])
-	if p.Err != nil {
+	startKey, err := timedTxKey(staker.StartTime(), priority, stakerID)
+	if err != nil {
 		// Close the DB, but ignore the error, as the parent error needs to be
 		// returned.
 		_ = prefixStartDB.Close()
-		return fmt.Errorf("couldn't serialize validator key: %w", p.Err)
+		return err
 	}
-	startKey := p.Bytes
 
 	errs := wrappers.Errs{}
 	errs.Add(
 		prefixStartDB.Put(startKey, txBytes),
 		prefixStartDB.Close(),
 	)
+
+	if priority != 1 {
```

Comment:
// If the staker being added is a validator, add to the pending validator database